File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataWeaponModel.java
package com.mcreater.amclcore.api.mihoyo.genshin.model;

import com.google.gson.annotations.SerializedName;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class GenshinGachaDataWeaponModel {
    private int id;
    private String name;
    private String icon;
    private int level;
    @SerializedName("cat_id")
    private int catId;
    @SerializedName("wiki_url")
    private String wikiUrl;
}

File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataWeaponModel.kt
package com.mcreater.amclcore.api.mihoyo.genshin.model

import com.google.gson.annotations.SerializedName

data class GenshinGachaDataWeaponModel(
    var id: Int = 0,
    var name: String? = null,
    var icon: String? = null,
    var level: Int = 0,
    @SerializedName("cat_id") var catId: Int = 0,
    @SerializedName("wiki_url") var wikiUrl: String? = null
)
